### ISSUE TITLE
Search not working; suggests using the desktop app

### ISSUE THREAD

**Issue by weeman1337** [vector-im/element-web#23628]
### Steps to reproduce

1. Try to search something

#### What did you expect?

Search should work

#### What happened instead?

Desktop app tells me to use the desktop app

![image](https://user-images.githubusercontent.com/6216686/198272241-4178b668-92f0-433e-984d-a57c9131a094.png)

Similar, closed issue  #17051

### Operating system

Ubuntu 22.04.1 LTS

### Application version

Element version: 1.11.12 Olm version: 3.2.12

### How did you install the app?

https://packages.element.io/debian/

### Homeserver

element.io

### Will you send logs?

Yes

**Comment by t3chguy** [comment]
```
2022-10-27T11:18:55.621Z I EventIndex: Event indexing is disabled, not initializing
```

Did you disable event indexing in settings?

**Comment by weeman1337** [comment]
> Did you disable event indexing in settings?

No.

Works after updating to 1.11.13 :man_shrugging:

**Comment by decadent** [comment]
Still occurs in 1.11.13 and 1.11.14 on Mac. The only thing changed on my part is that I recently updated to 1.11.13, and just now to 1.11.14 after encountering the bug.

I have no idea where I could disable ‘event indexing’, and I'm the type of guy who looks through all the settings after installing and then again occasionally. However, in the settings under ‘Security and privacy’ → ‘Message search’ I see this:

> Element is missing some components required for securely caching encrypted messages locally. If you'd like to experiment with this feature, build a custom Element Desktop with [search components added](https://github.com/vector-im/element-desktop/blob/develop/docs/native-node-modules.md#adding-seshat-for-search-in-e2e-encrypted-rooms).

So my guess is, the build process was changed to exclude those components, for some reason.

I installed Element with Homebrew, which downloads [the latest dmg archive from packages.riot.im](https://packages.riot.im/desktop/install/macos/Element.dmg).

**Comment by t3chguy** [comment]
> So my guess is, the build process was changed to exclude those components, for some reason.

That is not the case, I don't use the homebrew but install the dmg manually and rely on the built-in self-updater and it works fine.

**Comment by dkasak** [comment]
Another MacOS user reported seeing

> Element is missing some components required for securely caching encrypted messages locally. If you'd like to experiment with this feature, build a custom Element Desktop with [search components added](https://github.com/vector-im/element-desktop/blob/develop/docs/native-node-modules.md#adding-seshat-for-search-in-e2e-encrypted-rooms).

on Element Desktop 1.11.16. The error they're getting in the logs is

    2022-12-09T11:18:13.537Z I EventIndex: Event indexing isn't installed for the platform, not initializing.

which makes it sound like Seshat support is not there at all. They've installed Element Desktop from element.io.

The error is triggered [from here](https://github.com/matrix-org/matrix-react-sdk/blob/65f98435761e3a8d9ef6b95abe3d16a7f212716a/src/indexing/EventIndexPeg.ts#L60-L63). I find it a bit puzzling since the function in the condition simply checks `this._supportIsInstalled` and this is set just above from `indexManager.supportsEventIndexing()` which hardcodes returning `true`.

**Comment by lucajovine** [comment]
I have the same issue with Mac OS Desktop version 1.11.16. Launching from a shell, I get:

`> /Applications/Element.app/Contents/MacOS/Element
Seshat unexpected error: Error: dlopen(/Applications/Element.app/Contents/Resources/app.asar.unpacked/node_modules/matrix-seshat/native/index.node, 1): no suitable image found.  Did find:
	/Applications/Element.app/Contents/Resources/app.asar.unpacked/node_modules/matrix-seshat/native/index.node: cannot load 'index.node' (load command 0x80000034 is unknown)
	/Applications/Element.app/Contents/Resources/app.asar.unpacked/node_modules/matrix-seshat/native/index.node: cannot load 'index.node' (load command 0x80000034 is unknown)
    at process.func [as dlopen] (node:electron/js2c/asar_bundle:5:1810)
    at Module._extensions..node (node:internal/modules/cjs/loader:1205:18)
    at Object.func [as .node] (node:electron/js2c/asar_bundle:5:2037)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:829:12)
    at c._load (node:electron/js2c/asar_bundle:5:13339)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Applications/Element.app/Contents/Resources/app.asar/node_modules/matrix-seshat/lib/index.js:16:22)
    at Module._compile (node:internal/modules/cjs/loader:1120:14) {
  code: 'ERR_DLOPEN_FAILED'
}
/Users/luca/Library/Application Support/Element exists: yes
/Users/luca/Library/Application Support/Riot exists: no
Starting auto update with base URL: https://packages.element.io/desktop/update/
Update URL: https://packages.element.io/desktop/update/macos/?localVersion=1.11.16
Fetching translation json for locale: en_EN
Changing application language to en
Fetching translation json for locale: en
Resetting the UI components after locale change
Resetting the UI components after locale change
Changing application language to en
Fetching translation json for locale: en
Resetting the UI components after locale change`

**Comment by decadent** [comment]
Perhaps this could be related to the OS version on Mac—I, for one, am using version 10.14 ‘Mojave’.

**Comment by lucajovine** [comment]
> Perhaps this could be related to the OS version on Mac—I, for one, am using version 10.14 ‘Mojave’.

...but can you search messages in an encrypted room?

**Comment by lucajovine** [comment]
Actually, I think this is the culprit:

```
> cd /Applications/Element.app/Contents/Resources/app.asar.unpacked/node_modules/matrix-seshat/native
> ./index.node
-bash: ./index.node: cannot execute binary file: Exec format error
```

i.e. the binary is not for Mac OS, or at least does not run on my system.

**Comment by dkasak** [comment]
We got to the bottom of this: load command `0x80000034` mentioned during startup is a dynamic linker command `LC_DYLD_CHAINED_FIXUPS`. Support for this command got added at some point after macOS 10.13 (High Sierra). Since the Seshat (the message search component) binary makes use of this command, macOS is unable to load it on that platform.

Unfortunately, due to Apple dropping support for High Sierra, it's considered an unsupported platform by Element Web:

> Supported
> - ...
> - Desktop OSes means macOS, Windows, and Linux versions for desktop devices that are actively supported by the OS vendor and receive security updates

Thanks for helping us debug. Sorry it wasn't a happier ending!

**Comment by decadent** [comment]
By any chance, is there an archive of binaries of earlier versions?

**Comment by t3chguy** [comment]
> By any chance, is there an archive of binaries of earlier versions?

https://packages.element.io/desktop/install/macos/

The self updater will need disabling via config.json though

**Comment by decadent** [comment]
Thanks! The ‘1.11.2’ image actually has 1.11.15 for some reason, but 1.11.1 works—though I had to relogin and reimport the decryption keys due to downgrading, and now after a half-hour the search index still seems patchy, doesn't find all messages. Dunno if Element continues to index in the background, there's no particular CPU activity from it anymore.

**Comment by t3chguy** [comment]
Yes 1.11.2 getting corrupted due to a bad symlink is a known issue. Also why a few versions are missing after it. I should just delete 1.11.2 given that

**Comment by lucajovine** [comment]
@t3chguy can you please re-enable ([https://packages.element.io/desktop/install/macos/]) ?
I'm getting a 404 error "This object could not be viewed/You are not authorized to view this object"...
Thanks!

**Comment by t3chguy** [comment]
We moved to R2 which doesn't support indexes, if you were to go to packages.element.io and click through it'd take you to https://packages.element.io/desktop/install/macos/index.html which works

**Comment by Folling** [comment]
Happening to me on Linux, doesn't log anything about event indexing. 1.12.10

**Comment by vanillaicecreamcone** [comment]
this also is happening to me on version 1.12.13 on linux (ubuntu)

**Comment by Kermamrek** [comment]
Also happening for me on windows
